src/helm/benchmark/runner.py

```diff
@@ -185,4 +219,111 @@ def run_one(self, run_spec: RunSpec):
             json.dumps(list(map(asdict_without_nones, per_instance_stats)), indent=2),
         )
 
+        write(
+            os.path.join(run_path, "instances.json"),
+            json.dumps([asdict_without_nones(instance) for instance in scenario_state.instances], indent=2),
+        )
+
+        schema = read_schema()
```

Comment:
This function is getting a bit long, would be nice to move this into a separate function with a top-level docstring...

Reply:
Makes sense, I'll do it in a different PR.